### PR TITLE
media  modal activation

### PR DESCRIPTION
**Pull request by itkg-rboessel**

### CODE REVIEW

File: BackofficeBundle/Resources/public/coffee/block/wysiwyg.coffee [hunk 1]
```diff
@@ -0,0 +1,2 @@
+tinymce_button_mediamanager= (ed) ->
+  openMediaModal "select_media_modal"
```
Comment (nicolasThal) [inline BackofficeBundle/Resources/public/coffee/block/wysiwyg.coffee]: eof

File: BackofficeBundle/Resources/views/form.html.twig [hunk 2]
```diff
@@ -11,4 +11,9 @@
     <span class="orchestra-hidden-info" id="dynamic-modal-title">{{ ('open_orchestra_backoffice.block.' ~ blockType ~ '.title') | trans }}</span>
```
Comment (itkg-ngilain) [inline BackofficeBundle/Resources/views/form.html.twig]: Not for this line, but for the file above : BackofficeBundle/Resources/public/coffee/table/.tableviewLoader.coffee.swp is useless, it should be removed

**Comment by itkg-ngilain** [comment]
:+1: when you have removed the useless file

**Comment by itkg-nanne** [comment]
:+1:

File: MediaAdminBundle/Resources/public/coffee/orchestraMediaType.coffee [hunk 3]
```diff
@@ -1,8 +1,9 @@
 extendView = extendView || {}
-extendView['orchestraMediaType'] =
+extendView['OrchestraMediaType'] =
```
Comment (itkg-nanne) [inline MediaAdminBundle/Resources/public/coffee/orchestraMediaType.coffee]: It is a key so keep lowerCase

File: BackofficeBundle/Resources/config/form.yml [hunk 4]
```diff
@@ -46,11 +46,9 @@ parameters:
                     - searchreplace wordcount visualblocks visualchars code fullscreen
                     - insertdatetime media nonbreaking save table contextmenu directionality
                     - emoticons template paste textcolor
-                toolbar1: undo redo | styleselect | bold italic | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | link
+                toolbar1: undo redo | styleselect | bold italic | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | link mediamanager
                 toolbar2: print preview | forecolor backcolor
```
Comment (itkg-rboessel) [inline BackofficeBundle/Resources/config/form.yml]: i don't know how to override this line in Media bundle

**Comment by itkg-alavieille** [comment]
careful if you merguez after me, you will have conflicts with this PR https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/833

**Comment by itkg-alavieille** [comment]
My PR https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/833 is merge, so can you use the viewConfigurator when you create a new backbone view

**Comment by itkg-rboessel** [comment]
@itkg-ngilain @itkg-nanne @itkg-alavieille @nicolasThal lot of modifications since last +1

File: MediaAdminBundle/Resources/public/coffee/orchestraAbstractMediaType.coffee [hunk 5]
```diff
@@ -0,0 +1,59 @@
+extendView = extendView || {}
+extendView['orchestraMediaAbstractType'] =
+  events:
+    'click .clear-media': 'clearMedia'
```
Comment (itkg-nanne) [inline MediaAdminBundle/Resources/public/coffee/orchestraAbstractMediaType.coffee]: If you have no clear-media in tinyMce, this event should'nt be in this extendView.

File: MediaAdminBundle/Resources/public/coffee/orchestraAbstractMediaType.coffee [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        )
+      error: ->
+        new viewClass(
+          body: 'Erreur durant le chargement'
+          domContainer: modal
+          input: inputId
+        )
+    return
+
+callback_tinymce_init = null
```
Comment (itkg-nanne) [inline MediaAdminBundle/Resources/public/coffee/orchestraAbstractMediaType.coffee]: The extendViews could be in a extendView Folder.
The bind in another fiel too.

File: MediaAdminBundle/Resources/public/coffee/orchestraMediaType.coffee [hunk 7]
```diff
@@ -1,47 +1,13 @@
...
-          input: inputId
-        )
-    return
-
-widgetChannel.bind 'ready', (view) ->
-  if $("[data-prototype*='clear-media']", view.$el).length > 0
-    $.extend true, view, extendView['orchestraMediaType']
-    return view.delegateEvents()
-  return
+    @openMediaModal(modal, inputId, url, @method)
```
Comment (itkg-nanne) [inline MediaAdminBundle/Resources/public/coffee/orchestraMediaType.coffee]: why not call abstractOpenMediaModal: (modal, inputId, url, method)?
You don't need two methods abstractOpenMediaModal and OpenMediaModal?

File: MediaAdminBundle/DependencyInjection/Compiler/TinymceCompilerPass.php [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        if ($container->hasDefinition('twig.extension.stfalcon_tinymce')) {
+            $param = $container->getParameter('stfalcon_tinymce.config');
+            if ($container->hasParameter('open_orchestra_media_admin.tinymce')) {
+                $param = array_merge($param, $container->getParameter('open_orchestra_media_admin.tinymce'));
+            }
+            if (isset($param["theme"])
+                && isset($param["theme"]["simple"])
+                && isset($param["theme"]["simple"]["toolbar1"])) {
+                $param["theme"]["simple"]["toolbar1"] .= " mediamanager";
+                $param["theme"]["simple"]["toolbar1"] = str_replace(' image ', ' ', " " . $param["theme"]["simple"]["toolbar1"]);
```
Comment (itkg-ngilain) [inline MediaAdminBundle/DependencyInjection/Compiler/TinymceCompilerPass.php]: Do this mean that the OpenOrchestra Media Button can only be set on the 'simple' theme ?
If yes, maybe we should think about another way to introduce it, not to be dependant on a specific theme.

File: MediaAdminBundle/Resources/public/coffee/GalleryLoad.coffee [hunk 9]
```diff
@@ -1,4 +1,5 @@
-GalleryLoad = (link, target) ->
+GalleryLoad = (link, target, mediaView) ->
```
Comment (itkg-ngilain) [inline MediaAdminBundle/Resources/public/coffee/GalleryLoad.coffee]: Are you sure of the order of the params ?
target can be undefined. Is it possible to call the GalleryLoad with no target defined and a mediaView defined? In that case, the mediaView will be received not as the mediaView, but as the target.

File: MediaAdminBundle/Resources/public/coffee/wysiwyg/MediaModalWysiwygView.coffee [hunk 10]
```diff
@@ -0,0 +1,8 @@
+currentModal = null
+
+MediaModalWysiwygView = MediaModalView.extend(
+
```
Comment (nicolasThal) [inline MediaAdminBundle/Resources/public/coffee/wysiwyg/MediaModalWysiwygView.coffee]: empty line to suppress

File: MediaAdminBundle/Resources/public/coffee/extendView/orchestraAbstractMediaType.coffee [hunk 11]
```diff
@@ -10,14 +8,10 @@ extendView['orchestraMediaType'] =
...
-    event.preventDefault()
-    @method = if @options.method then @options.method else 'GET'
-    target = $(event.currentTarget)
-    modal = $('#' + target.data("target"), @$el)
-    inputId = target.data("input")
-    url = target.data("url")
-    viewClass = appConfigurationView.getConfiguration('media', 'showMediaModal')
+
+  openMediaModal: (modal, inputId, url, method, mediaModalView) ->
+    mediaModalView = mediaModalView or "showMediaModal"
```
Comment (itkg-ngilain) [inline MediaAdminBundle/Resources/public/coffee/extendView/orchestraAbstractMediaType.coffee]: Can you use instead the feature of coffee allowing to declare default value in function signature ?
openMediaModal: (modal, inputId, url, method, mediaModalView = "showMediaModal")

File: MediaAdminBundle/DependencyInjection/Compiler/TinymceCompilerPass.php (hunk 8 above)
Comment (nicolasThal) [inline MediaAdminBundle/DependencyInjection/Compiler/TinymceCompilerPass.php]: For a basic utilisation, I don't think so, but we can create a ticket to improve the way the configuration is added to tinymce

File: MediaAdminBundle/Resources/public/coffee/GalleryLoad.coffee [hunk 12]
```diff
@@ -1,4 +1,5 @@
...
+  mediaView = mediaView or "showGalleryCollection";
```
Comment (itkg-ngilain) [inline MediaAdminBundle/Resources/public/coffee/GalleryLoad.coffee]: After a second read, I'll change the parameters order and add a coffee default value to make something like this :
GalleryLoad = (link, mediaView = "showGalleryCollection", target) ->
But beware to report the modification on GalleryLoad call

File: MediaAdminBundle/Resources/public/coffee/bindExtendView.coffee [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+  isMediaView = $("[data-prototype*='clear-media']", view.$el).length > 0
+  isWysiwygView = $(".select_media_modal", view.$el).length > 0
+  if isMediaView or isWysiwygView
+    $.extend true, view, extendView['orchestraMediaAbstractType']
+    if isMediaView
+      $.extend true, view, extendView['orchestraMediaType']
+    if isWysiwygView
+      do (view) ->
+        callback_tinymce_init = (editor) ->
+          $.extend true, view, extendView['orchestraWysiwygType']
```
Comment (itkg-nanne) [inline MediaAdminBundle/Resources/public/coffee/bindExtendView.coffee]: you should send view.delegatesEvents here in order to bind in backbone view.
In this way, you could bin in extend view on [aria-label="mediamanager"] and not use on('click')

**Comment by itkg-nanne** [comment]
:+1:

**Pull request merged**